api/src/main/java/org/apache/iceberg/actions/RewritePositionDeleteFiles.java

```diff
@@ -41,11 +119,60 @@
   RewritePositionDeleteFiles filter(Expression expression);
 
   /** The action result that contains a summary of the execution. */
+  @Value.Immutable
   interface Result {
+    List<PositionDeleteGroupRewriteResult> rewriteResults();
+
     /** Returns the count of the position deletes that been rewritten. */
     int rewrittenDeleteFilesCount();
 
     /** Returns the count of the added delete files. */
     int addedDeleteFilesCount();
+
+    /** Returns the number of bytes of position deletes that have been rewritten */
+    long rewrittenBytesCount();
+
+    /** Returns the number of bytes of newly added position deletes */
+    long addedBytesCount();
+  }
+
+  /**
+   * For a particular position delete file group, the number of position delete files which are
+   * newly created and the number of files which were formerly part of the table but have been
+   * rewritten.
+   */
+  @Value.Immutable
+  interface PositionDeleteGroupRewriteResult {
```

Comment:
Since it is a nested class, what about simply `FileGroupRewriteResult` and `FileGroupInfo`? We usually rarely add any context in the name for nested classes. Not a strong opinion, though. Would love to hear more from others.

Reply:
Immutables generate all of these into the same package (org.apache.iceberg.actions...), so this collides with the RewriteFile one (although technically, it is the same object..)

Reply:
Did you add the required annotation on the outer class? I forgot the name of it.

Reply:
Sorry , not familiar, if you can point me to that

Reply:
Oh I see, well I think the problem is not that, they're no enclosing each other.  it's just they all end up in the same place.  That being said, FileGroupRewriteResult seems to me a bit particular to RewriteDataFiles somehow

Reply:
I see, used that annotation and it does not collide anymore.  Made the names match RewriteDataFiles.Result classes